### ISSUE TITLE
Docker's ~/.docker/config.json file is reset upon Windows application restart

### ISSUE THREAD

**Issue by MarkHarrisCanterbury** [docker/for-win#13529]
### Description

When the Docker desktop application is restarted in Windows the .docker/config.json file is overwritten, so that the "credsStore" key's value is changed to "desktop.exe".

### Reproduce

1. Start Ubuntu WSL and Docker Desktop app in Windows
2. In WSL, change the ~/.docker/config.json file's "credsStore" key value to "ecr-login"
3. Restart Docker Desktop app
4. ~/.docker/config.json file's "credsStore" key value is changed to "desktop.exe"

### Expected behavior

When the Docker desktop application is restarted in Windows, the .docker/config.json file is not changed.

### docker version

```bash
Client:
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.1
 Git commit:        20.10.21-0ubuntu1~22.04.3
 Built:             Thu Apr 27 05:57:17 2023
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop
 Engine:
  Version:          24.0.2
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       659604f
  Built:            Thu May 25 21:52:17 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.5)
  compose: Docker Compose (Docker Inc., v2.18.1)
  dev: Docker Dev Environments (Docker Inc., v0.1.0)
  extension: Manages Docker extensions (Docker Inc., v0.2.19)
  init: Creates Docker-related starter files for your project (Docker Inc., v0.1.0-beta.4)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.26.0)
  scout: Command line tool for Docker Scout (Docker Inc., v0.12.0)

Server:
 Containers: 27
  Running: 27
  Paused: 0
  Stopped: 0
 Images: 46
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 5.15.90.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.47GiB
 Name: docker-desktop
 ID: 560b6de5-77ec-4677-b127-96d39655bc8b
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Diagnostics ID

FB6DE0E8-61EA-4B62-B718-29D081250211/20230606031133

**Comment by 6611152275** [comment]
> ### Description
> When the Docker desktop application is restarted in Windows the .docker/config.json file is overwritten, so that the "credsStore" key's value is changed to "desktop.exe".
> 
> ### Reproduce
> 1. Start Ubuntu WSL and Docker Desktop app in Windows
> 2. In WSL, change the ~/.docker/config.json file's "credsStore" key value to "ecr-login"
> 3. Restart Docker Desktop app
> 4. ~/.docker/config.json file's "credsStore" key value is changed to "desktop.exe"
> 
> ### Expected behavior
> When the Docker desktop application is restarted in Windows, the .docker/config.json file is not changed.
> 
> ### docker version
> ```shell
> Client:
>  Version:           20.10.21
>  API version:       1.41
>  Go version:        go1.18.1
>  Git commit:        20.10.21-0ubuntu1~22.04.3
>  Built:             Thu Apr 27 05:57:17 2023
>  OS/Arch:           linux/amd64
>  Context:           default
>  Experimental:      true
> 
> Server: Docker Desktop
>  Engine:
>   Version:          24.0.2
>   API version:      1.43 (minimum version 1.12)
>   Go version:       go1.20.4
>   Git commit:       659604f
>   Built:            Thu May 25 21:52:17 2023
>   OS/Arch:          linux/amd64
>   Experimental:     false
>  containerd:
>   Version:          1.6.21
>   GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
>  runc:
>   Version:          1.1.7
>   GitCommit:        v1.1.7-0-g860f061
>  docker-init:
>   Version:          0.19.0
>   GitCommit:        de40ad0
> ```
> 
> ### docker info
> ```shell
> Client:
>  Context:    default
>  Debug Mode: false
>  Plugins:
>   buildx: Docker Buildx (Docker Inc., v0.10.5)
>   compose: Docker Compose (Docker Inc., v2.18.1)
>   dev: Docker Dev Environments (Docker Inc., v0.1.0)
>   extension: Manages Docker extensions (Docker Inc., v0.2.19)
>   init: Creates Docker-related starter files for your project (Docker Inc., v0.1.0-beta.4)
>   sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
>   scan: Docker Scan (Docker Inc., v0.26.0)
>   scout: Command line tool for Docker Scout (Docker Inc., v0.12.0)
> 
> Server:
>  Containers: 27
>   Running: 27
>   Paused: 0
>   Stopped: 0
>  Images: 46
>  Server Version: 24.0.2
>  Storage Driver: overlay2
>   Backing Filesystem: extfs
>   Supports d_type: true
>   Using metacopy: false
>   Native Overlay Diff: true
>   userxattr: false
>  Logging Driver: json-file
>  Cgroup Driver: cgroupfs
>  Cgroup Version: 1
>  Plugins:
>   Volume: local
>   Network: bridge host ipvlan macvlan null overlay
>   Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
>  Swarm: inactive
>  Runtimes: io.containerd.runc.v2 runc
>  Default Runtime: runc
>  Init Binary: docker-init
>  containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
>  runc version: v1.1.7-0-g860f061
>  init version: de40ad0
>  Security Options:
>   seccomp
>    Profile: builtin
>  Kernel Version: 5.15.90.1-microsoft-standard-WSL2
>  Operating System: Docker Desktop
>  OSType: linux
>  Architecture: x86_64
>  CPUs: 8
>  Total Memory: 15.47GiB
>  Name: docker-desktop
>  ID: 560b6de5-77ec-4677-b127-96d39655bc8b
>  Docker Root Dir: /var/lib/docker
>  Debug Mode: false
>  HTTP Proxy: http.docker.internal:3128
>  HTTPS Proxy: http.docker.internal:3128
>  No Proxy: hubproxy.docker.internal
>  Registry: https://index.docker.io/v1/
>  Labels:
>  Experimental: false
>  Insecure Registries:
>   hubproxy.docker.internal:5555
>   127.0.0.0/8
>  Live Restore Enabled: false
> ```
> 
> ### Diagnostics ID
> FB6DE0E8-61EA-4B62-B718-29D081250211/20230606031133
> 
> ### Additional Info
> _No response_

**Comment by 6611152275** [comment]
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.5)
  compose: Docker Compose (Docker Inc., v2.18.1)
  dev: Docker Dev Environments (Docker Inc., v0.1.0)
  extension: Manages Docker extensions (Docker Inc., v0.2.19)
  init: Creates Docker-related starter files for your project (Docker Inc., v0.1.0-beta.4)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.26.0)
  scout: Command line tool for Docker Scout (Docker Inc., v0.12.0)

Server:
 Containers: 27
  Running: 27
  Paused: 0
  Stopped: 0
 Images: 46
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 5.15.90.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.47GiB
 Name: docker-desktop
 ID: 560b6de5-77ec-4677-b127-96d39655bc8b
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by MihaelaStoica** [comment]
This issue has been fixed in Docker Desktop 4.27. See [release note](https://docs.docker.com/desktop/release-notes/#:~:text=Fixed%20a%20bug%20where%20the%20credentials%20store%20configuration%20would%20reset%20on%20app%20restart%20with%20the%20WSL%20integration.%20Fixes%20docker/for%2Dwin%2313529.).
